common/middleware/updateCorrelationId.js

```diff
@@ -0,0 +1,21 @@
+// checks if there is a keycloak request id in the headers and uses that for correlation ID if found
+const cls = require('cls-hooked')
+const { clsNamespace } = require('../config')
+const { updateMDC } = require('../utils/util')
+
+const updateCorrelationId = (req, res, next) => {
+  const {
+    headers: { 'x-request-id': correlationId },
+  } = req
```

Comment:
Could be extracted in the declaration:
`const updateCorrelationId = ( { headers: { 'x-request-id': correlationId } }, res, next) => {`